OpenEdXMobile/src/main/java/org/edx/mobile/view/TabsBaseFragment.java

```diff
@@ -48,21 +49,38 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        // Handle tab-selection through deep link (if available)
-        if (getArguments() != null && binding != null) {
-            @ScreenDef String screenName = getArguments().getString(Router.EXTRA_SCREEN_NAME);
+        handleTabSelection(getArguments());
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleTabSelection(intent.getExtras());
+    }
+
+    /**
+     * Method to handle the tab-selection of the viewpager based on screen name {@link Screen}.
```

Comment:
Method to handle the tab-selection of the ViewPager based on screen name {@link Screen} which may be sent through a deep link.